ansi/graphics.go

```diff
@@ -8,11 +8,65 @@ import (
 	"image"
 	"io"
 	"os"
+	"strconv"
 	"strings"
 
 	"github.com/charmbracelet/x/ansi/kitty"
+	"github.com/charmbracelet/x/ansi/sixel"
 )
 
+// SixelGraphics returns a sequence that encodes the given sixel image payload to
+// a DCS sixel sequence.
+//
+//	DCS p1; p2; p3; q [sixel payload] ST
+//
+// p1 = pixel aspect ratio, deprecated and replaced by pixel metrics in the payload
+//
+// p2 = This is supposed to be 0 for transparency, but terminals don't seem to
+// to use it properly. Value 0 leaves an unsightly black bar on all terminals
+// I've tried and looks correct with value 1.
+//
+// p3 = Horizontal grid size parameter. Everyone ignores this and uses a fixed grid
+// size, as far as I can tell.
+//
+// See https://shuford.invisible-island.net/all_about_sixels.txt
+func SixelGraphics(p1, p2, p3 int, payload []byte) string {
+	var buf bytes.Buffer
+
+	buf.WriteString("\x1bP")
+	if p1 >= 0 {
+		buf.WriteString(strconv.Itoa(p1))
+	}
+	buf.WriteByte(';')
+	if p2 >= 0 {
+		buf.WriteString(strconv.Itoa(p2))
+	}
+	buf.WriteByte(';')
+	if p3 >= 0 {
+		buf.WriteString(strconv.Itoa(p3))
+	}
+	buf.WriteString(";q")
+	buf.Write(payload)
+	buf.WriteString("\x1b\\")
+
+	return buf.String()
+}
+
+// WriteSixelGraphics encodes an image as sixels into the provided io.Writer.
+// Options is provided in expectation of future options (such as dithering), but
+// none are yet implemented. o can be nil to use the default options.
+func WriteSixelGraphics(w io.Writer, m image.Image) error {
```

Comment:
I think "Encode" signifies the meaning more

```suggestion
// EncodeSixelGraphics encodes an image as sixels into the provided io.Writer.
// Options is provided in expectation of future options (such as dithering), but
// none are yet implemented. o can be nil to use the default options.
func EncodeSixelGraphics(w io.Writer, m image.Image) error {
```

Reply:
I wonder if we actually need this function 🤔

Implementors can use the `sixel` package and write the encoded output in a `ansi.SixelGraphics`. This way they can encode the images with whatever encoding options we have.

Reply:
yea I would rather remove this function